### PR TITLE
Feature/create folder

### PR DESCRIPTION
**Pull request by avivbarel**
Added folder creation, tests included.
Moved authentication check to request handler to avoid repetition.
Issue #68

### CODE REVIEW

File: upload/upload.go [hunk 1]
```diff
@@ -119,6 +122,17 @@ func (r *Router) Setup(rg *gin.RouterGroup) {
 
 // Upload is the request handler for /upload request.
 func (r *Router) Upload(c *gin.Context) {
+	reqUser := user.ExtractRequestUser(c)
```
Comment (omerzamir) [inline upload/upload.go]: after checking it, pass it as a parameter to every func.

File: upload/upload.go [hunk 2]
```diff
@@ -138,20 +152,58 @@ func (r *Router) Upload(c *gin.Context) {
...
+	return fileName
+}
+
+// UploadFolder creates a folder in file service
+func (r *Router) UploadFolder(c *gin.Context) {
+	reqUser := user.ExtractRequestUser(c)
+
+	folderFullName := getFileName(c)
+	if folderFullName == "" {
+		folderFullName = uuid.NewV4().String()
```
Comment (omerzamir) [inline upload/upload.go]: return error

File: upload/upload.go [hunk 3]
```diff
@@ -138,20 +152,58 @@ func (r *Router) Upload(c *gin.Context) {
...
+	}
+
+	createFolderResp, err := r.fileClient.CreateFile(c.Request.Context(), &fpb.CreateFileRequest{
+		Key:     "",
+		Bucket:  "bucket is required currently, so i gotta fill this string",
```
Comment (omerzamir) [inline upload/upload.go]: set it to reqUser.ID

File: upload/upload.go [hunk 4]
```diff
@@ -119,6 +122,17 @@ func (r *Router) Setup(rg *gin.RouterGroup) {
 
 // Upload is the request handler for /upload request.
 func (r *Router) Upload(c *gin.Context) {
+	reqUser := user.ExtractRequestUser(c)
+	if reqUser == nil {
+		c.AbortWithStatus(http.StatusUnauthorized)
```
Comment (omerzamir) [inline upload/upload.go]: change it to AbortWithError and generate your own error with fmt

File: upload/upload.go [hunk 5]
```diff
@@ -138,20 +152,58 @@ func (r *Router) Upload(c *gin.Context) {
 	}
 }
 
+func getFileName(c *gin.Context) string {
```
Comment (AllKinds) [inline upload/upload.go]: add function documentation

File: upload/upload.go [hunk 6]
```diff
@@ -138,20 +152,58 @@ func (r *Router) Upload(c *gin.Context) {
...
+	fileName := ""
+	contentDisposition := c.GetHeader(ContentDispositionHeader)
+
+	if contentDisposition != "" {
+		_, err := fmt.Sscanf(contentDisposition, "filename=%s", &fileName)
```
Comment (AllKinds) [inline upload/upload.go]: Dont you have to add a space between `contentDisposition` and the next string?

Comment (AllKinds) [inline upload/upload.go]: Dont you have to add a space between `contentDisposition` and the next string?

File: upload/upload.go (hunk 1 above)
Comment (omerzamir) [reply upload/upload.go]: Issue will be opened to resolve this comment.
**Out of context.**

File: upload/upload.go [hunk 7]
```diff
@@ -138,20 +153,60 @@ func (r *Router) Upload(c *gin.Context) {
...
+	return fileName
+}
+
+// UploadFolder creates a folder in file service
+func (r *Router) UploadFolder(c *gin.Context) {
+	reqUser := user.ExtractRequestUser(c)
+
+	folderFullName := extractFileName(c)
+	if folderFullName == "" {
+		err := c.AbortWithError(http.StatusBadRequest, fmt.Errorf("Folder name not specified"))
```
Comment (omerzamir) [inline upload/upload.go]: error message should start with lowercase character, we didn't mention it earlier so ill fix it for you and merge.
also, it should be implemented like line 194, ill fix it too in the same commit.

**Pull request merged**